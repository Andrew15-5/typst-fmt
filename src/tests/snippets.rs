use super::*;

make_test!(code_func, "#{f(1,2,3)}");
make_test!(
    code_func_break,
    "#{f(1,2,3)}",
    Config {
        max_line_length: 2,
        ..Default::default()
    }
);
make_test!(
    code_func_break_nested,
    "#{{f(1,2,3)}}",
    Config {
        max_line_length: 2,
        ..Default::default()
    }
);
make_test!(while_loop, WHILE_LOOP);
make_test!(for_loop, FOR_LOOP);

const FOR_LOOP: &str = r#"#for k in range(5) {
    repr(k) + " " 
}"#;

const WHILE_LOOP: &str = r#"#let i = 0
#while true {
  i += 1
  if i > 15 { break }
  repr(i) + " "
}"#;
