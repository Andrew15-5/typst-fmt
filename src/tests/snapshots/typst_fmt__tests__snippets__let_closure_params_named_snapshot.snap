---
source: src/tests/snippets.rs
description: "INPUT\n===\n\"#let hlinex(\\n  start: 0, end: auto, y  : auto,\\n\\n  stroke: auto,\\n  stop-pre-gutter: auto, gutter-restrict: none,\\n  \\n  \\n  stroke-expand: true,\\n  expand: none\\n) = (\\n  tablex-dict-type: \\\"hline\\\",\\n  start: start,\\n  end : end,\\n  y:y  ,\\n  stroke: stroke, \\n  parent: none\\n)\"\n===\n#let hlinex(\n  start: 0, end: auto, y  : auto,\n\n  stroke: auto,\n  stop-pre-gutter: auto, gutter-restrict: none,\n  \n  \n  stroke-expand: true,\n  expand: none\n) = (\n  tablex-dict-type: \"hline\",\n  start: start,\n  end : end,\n  y:y  ,\n  stroke: stroke, \n  parent: none\n)\n===\nFORMATTED\n===\n#let hlinex(\n  start: 0,\n  end: auto,\n  y: auto,\n  stroke: auto,\n  stop-pre-gutter: auto,\n  gutter-restrict: none,\n  stroke-expand: true,\n  expand: none,\n) = (\n  tablex-dict-type: \"hline\",\n  start: start,\n  end: end,\n  y: y,\n  stroke: stroke,\n  parent: none,\n)"
expression: formatted
---
"#let hlinex(\n  start: 0,\n  end: auto,\n  y: auto,\n  stroke: auto,\n  stop-pre-gutter: auto,\n  gutter-restrict: none,\n  stroke-expand: true,\n  expand: none,\n) = (\n  tablex-dict-type: \"hline\",\n  start: start,\n  end: end,\n  y: y,\n  stroke: stroke,\n  parent: none,\n)"
